testnet/stacks-node/src/tests/signer.rs

```diff
@@ -1098,6 +1110,107 @@ fn stackerdb_sign() {
     info!("Sign Time Elapsed: {:.2?}", sign_elapsed);
 }
 
+#[test]
+#[ignore]
+/// Test that a signer can be offline when a DKG round has commenced and
+/// can rejoin the DKG round after it has restarted
+fn stackerdb_delayed_dkg() {
+    if env::var("BITCOIND_TEST") != Ok("1".into()) {
+        return;
+    }
+
+    tracing_subscriber::registry()
+        .with(fmt::layer())
+        .with(EnvFilter::from_default_env())
+        .init();
+
+    info!("------------------------- Test Setup -------------------------");
+    let timeout = Duration::from_secs(200);
+    let mut signer_test = SignerTest::new(3);
+    boot_to_epoch_3_reward_set_calculation_boundary(
+        &signer_test.running_nodes.conf,
+        &signer_test.running_nodes.blocks_processed,
+        &signer_test.signer_stacks_private_keys,
+        &signer_test.signer_stacks_private_keys,
+        &mut signer_test.running_nodes.btc_regtest_controller,
+    );
+    let reward_cycle = signer_test.get_current_reward_cycle().saturating_add(1);
+    let public_keys = signer_test.get_signer_public_keys(reward_cycle);
+    let coordinator_selector = CoordinatorSelector::from(public_keys);
+    let (_, coordinator_public_key) = coordinator_selector.get_coordinator();
+    let coordinator_public_key =
+        StacksPublicKey::from_slice(coordinator_public_key.to_bytes().as_slice()).unwrap();
+
+    info!("------------------------- Stop Signers -------------------------");
+    let mut to_stop = None;
+    for (idx, key) in signer_test.signer_stacks_private_keys.iter().enumerate() {
+        let public_key = StacksPublicKey::from_private(key);
+        if public_key == coordinator_public_key {
+            // Do not stop the coordinator. We want coordinator to start a DKG round
+            continue;
+        }
+        // Only stop one signer
+        to_stop = Some(idx);
+        break;
+    }
+    let signer_idx = to_stop.expect("Failed to find a signer to stop");
+    let signer_key = signer_test.stop_signer(signer_idx);
+    debug!(
+        "Removed signer {signer_idx} with key: {:?}, {}",
+        signer_key,
+        signer_key.to_hex()
+    );
+
+    info!("------------------------- Start DKG -------------------------");
+    let height = signer_test
+        .running_nodes
+        .btc_regtest_controller
+        .get_headers_height();
+    // Advance one more to trigger DKG
+    run_until_burnchain_height(
+        &mut signer_test.running_nodes.btc_regtest_controller,
+        &signer_test.running_nodes.blocks_processed,
+        height.wrapping_add(1),
+        &signer_test.running_nodes.conf,
+    );
+    // Wait a bit so DKG is actually triggered and signers are not available to respond
+    std::thread::sleep(Duration::from_secs(5));
+
+    // Make sure DKG did not get set
+    assert!(signer_test
+        .stacks_client
+        .get_approved_aggregate_key(reward_cycle)
+        .expect("Failed to get approved aggregate key")
+        .is_none());
```

Comment:
Won't this never happen, even if you didn't stop a signer, because you haven't advanced the Stacks tip? So the `vote-for-aggregate-key` votes wouldn't be confirmed here?

Also because, if I'm reading this right, we're still in 2.5, so there wouldn't be Nakamoto blocks.

Reply:
Oh good point. I should mine one block before to prove that DKG did not proceed. Will add. And yes this is 2.5. I don't make any references to Nakamoto blocks in this test for that reason.